internal/crypto/cert.go

```diff
@@ -179,60 +58,6 @@ func (ca *CA) MakeAndWriteServerCert(certFile, keyFile string, hostnames []strin
 	return server, nil
 }
 
-type CertificateExtensionFunc func(*x509.Certificate) error
-
-func (ca *CA) MakeServerCert(hostnames []string, expiryDays int, fns ...CertificateExtensionFunc) (*TLSCertificateConfig, error) {
-	if len(hostnames) < 1 {
-		return nil, fmt.Errorf("at least one hostname must be provided")
-	}
-
-	serverPublicKey, serverPrivateKey, publicKeyHash, _ := NewKeyPairWithHash()
-
-	now := time.Now()
-	serverTemplate := &x509.Certificate{
-		Subject: pkix.Name{CommonName: hostnames[0]},
-
-		SignatureAlgorithm: x509.ECDSAWithSHA256,
-
-		NotBefore:    now.Add(-1 * time.Second),
-		NotAfter:     now.Add(time.Duration(expiryDays) * 24 * time.Hour),
-		SerialNumber: big.NewInt(1),
-
-		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
-		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
-		BasicConstraintsValid: true,
-
-		AuthorityKeyId: ca.Config.Certs[0].SubjectKeyId,
-		SubjectKeyId:   publicKeyHash,
-	}
-	serverTemplate.IPAddresses, serverTemplate.DNSNames = oscrypto.IPAddressesDNSNames(hostnames)
-	for _, fn := range fns {
-		if err := fn(serverTemplate); err != nil {
-			return nil, err
-		}
-	}
-
-	serverCrt, err := ca.signCertificate(serverTemplate, serverPublicKey)
-	if err != nil {
-		return nil, err
-	}
-	server := &TLSCertificateConfig{
-		Certs: append([]*x509.Certificate{serverCrt}, ca.Config.Certs...),
-		Key:   serverPrivateKey,
-	}
-	return server, nil
-}
-
-func (ca *CA) signCertificate(template *x509.Certificate, requestKey crypto.PublicKey) (*x509.Certificate, error) {
-	// Increment and persist serial
-	serial, err := ca.SerialGenerator.Next(template)
-	if err != nil {
-		return nil, err
-	}
-	template.SerialNumber = big.NewInt(serial)
-	return signCertificate(template, requestKey, ca.Config.Certs[0], ca.Config.Key)
-}
-
 func (ca *CA) EnsureClientCertificate(certFile, keyFile string, subjectName string, expireDays int) (*TLSCertificateConfig, bool, error) {
```

Comment:
Same question as above

Reply:
Same answer - it does not touch private keys and cannot be expressed as a CSR->Sign operation in the later phases.